### ISSUE TITLE
ViewScoped Unload happening on file download

### ISSUE THREAD

**Issue by melloware** [omnifaces/omnifaces#511]
I have an h:outputLink which hits a REST Service and does a file download becuase the REST Service returns a content-disposition header.

```xml
<h:outputLink styleClass="ui-icon-export-xml" value="#{request.contextPath}/rest/public/download/xml/jcmramp/?view=#{rampViewerForm.viewType}&amp;location=#{rampViewerForm.locationType}">
</h:outputLink>
```

When  I hit this link and my file downloads now no matter what I do I get a ViewExpired exception.

So I notice this is a POST to the server ..
![image](https://user-images.githubusercontent.com/4399574/57726908-2c13a880-765e-11e9-8995-24099f30101b.png)

Am I doing something wrong this worked in JSF 2.1 with OmniFaces 2.0.  But we are now on JSF 2.2 with OmniFaces 2.7.1 and seeing this behavior.

Any advice or is this a bug?

**Comment by melloware** [comment]
More information its doing a POST first to the viewer.JSF then doing the proper GET of the h:outputlink.

![image](https://user-images.githubusercontent.com/4399574/57727677-1acb9b80-7660-11e9-89e2-6c5f700efa1b.png)

I don't know why it would be doing the POST first which is causing OmniFaces to then make the viewstate expired.

**Comment by melloware** [comment]
Verified its even happening if I don't use JSF component at all and just use an HTML anchor tag.

```xml
<a href="#{request.contextPath}/rest/public/download/xml/jcmramp/?view=#{rampViewerForm.viewType}&amp;location=#{rampViewerForm.locationType}&amp;timezone=#{jcmUserPreferences.timezone}" class="ui-icon-export-xml"></a>
```

OmniFaces is doing a POST first.

**Comment by melloware** [comment]
Yuck this is the real problem: https://stackoverflow.com/questions/7650063/how-can-i-prevent-window-onbeforeunload-from-being-triggered-by-javascript-href

**Comment by melloware** [comment]
For those who care I solved it with adding an  `onmousedown="OmniFaces.Unload.disable();"` to the link like this..

```xml
<h:outputLink  value="#{request.contextPath}/rest/public/download/xml/jcmramp"
                          onmousedown="OmniFaces.Unload.disable();">
</h:outputLink>
```

**Comment by BalusC** [comment]
This is expected behavior. But I notice that this is undocumented. I'll have to update the javadoc.

**Comment by BalusC** [comment]
I found an interesting observation. It also works fine when you just add the `download` attribute to the link.

```
<a href="file.ext" download="file.ext">download</a>
```

I noticed this when I wanted to add some JavaScript which automatically disables the unload event to links having a `download` attribute. But it thus already works natively .. So I'm not going to make changes to unload source code. I'll document the two ways to solve this problem:

1. Explicitly specify the `download` attribute. The (decent) webbrowser will then not anymore trigger the unload event.
2. Use `onclick="OmniFaces.Unload.disable()`.
